src/mqtt_client/mod.rs

```diff
@@ -302,14 +388,12 @@ where
         if !self.mqtt.client().is_connected() {
```

Comment:
Then invert the `if` arms.
But doesn't this cause a lot of traffic through the `process_event(sm::Events::Connected)`. Maybe that's fine though.